### DISCUSSION TITLE
Making visual difference between changed and added/removed lines

### DISCUSSION THREAD

**Discussion by sskras** [dandavison/delta#1775] [Q&A]
Thanks for the useful tool. I am newbie, and use it to highlight unified diffs.

I see that by default `delta` colorizes background for all changed lines. With `--diff-highlight` however no background seems to be colorized. Eg. comparing my Proxmox VM configurations:

<img src="https://github.com/user-attachments/assets/b12a0700-39a0-4148-993b-c9aa3e4a0f2b" width="49%" />
<img src="https://github.com/user-attachments/assets/dd031089-9d1f-41c4-8ca3-eaf22c3d68e6" width="49%" />
<p />

I would like some hybrid solution, however. Eg. by colorizing background only of the added/removed lines (in default mode). 

Or at least to add coloring only from the start to the end of the line (maybe including the initial `-` or `+`).
Eg. mock up for the `--diff-highlight` mode:

![image](https://github.com/user-attachments/assets/d6159bb0-3db6-467c-95e7-ea6011aabc22)

Is there any possibilities already built into the tool, or some issue created about that, or should I file an one?

**Comment by sskras** [comment]
PS. Oh, I am running `delta 0.17.0` built on Debian 12 using [**pkgsrc**](https://github.com/NetBSD/pkgsrc).

**Comment by dandavison** [comment]
Hi @sskras, `--diff-highlight` is specifically for emulating a perl script known as "diff highlight" that is distributed with git.

```
$ delta --help
...
      --diff-highlight
          Emulate diff-highlight.

          (https://github.com/git/git/tree/master/contrib/diff-highlight)
```

Delta has many options for styling diffs, use `delta --help` to view the full manual, but here are some entries from the short help:

```
$ delta -h | rg --no-line-number --json -A1 '(minus|plus|zero)' | delta
      --keep-plus-minus-markers
          Prefix added/removed lines with a +/- character, as git does
      --line-numbers-minus-style <STYLE>
          Style string for line numbers in the old (minus) version of the file
          [default: auto]
      --line-numbers-plus-style <STYLE>
          Style string for line numbers in the new (plus) version of the file
          [default: auto]
      --line-numbers-zero-style <STYLE>
          Style string for line numbers in unchanged (zero) lines [default:
          auto]
      --minus-empty-line-marker-style <STYLE>
          Style string for removed empty line marker [default: "normal auto"]
      --minus-emph-style <STYLE>
          Style string for emphasized sections of removed lines [default:
      --minus-non-emph-style <STYLE>
          Style string for non-emphasized sections of removed lines that have
          an emphasized section [default: minus-style]
      --minus-style <STYLE>
          Style string for removed lines [default: "normal auto"]
      --plus-emph-style <STYLE>
          Style string for emphasized sections of added lines [default: "syntax
      --plus-empty-line-marker-style <STYLE>
          Style string for added empty line marker [default: "normal auto"]
      --plus-non-emph-style <STYLE>
          Style string for non-emphasized sections of added lines that have an
          emphasized section [default: plus-style]
      --plus-style <STYLE>
          Style string for added lines [default: "syntax auto"]
      --zero-style <STYLE>
          Style string for unchanged lines [default: "syntax normal"]```
